client/cli/src/logging.rs

```diff
@@ -0,0 +1,336 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use ansi_term::{Colour, Style};
+use std::fmt;
+use std::fmt::Write as _;
+use std::iter;
+use tracing::{
+	span::{self, Attributes},
+	Event, Id, Level, Subscriber,
+};
+use tracing_log::NormalizeEvent;
+use tracing_subscriber::{
+	fmt::{
+		time::{FormatTime, SystemTime},
+		FmtContext, FormatEvent, FormatFields,
+	},
+	layer::Context,
+	registry::LookupSpan,
+	Layer,
+};
+
+pub(crate) struct EventFormat<T = SystemTime> {
+	pub(crate) timer: T,
+	pub(crate) ansi: bool,
+	pub(crate) display_target: bool,
+	pub(crate) display_level: bool,
+	pub(crate) display_thread_name: bool,
+}
+
+// NOTE: the following code has been inspiref from tracing-subscriber
```

Comment:
small nit, maybe
// NOTE: the following code was inspired by tracing-subscriber
or
// NOTE: the following code took inspiration from tracing-subscriber

Reply:
Done in 2401a7ca1